pkg/prometheus/statefulset.go

```diff
@@ -701,17 +701,21 @@ func makeStatefulSetSpec(p monitoringv1.Prometheus, c *Config, ruleConfigMapName
 		if p.Spec.Thanos.Image != nil && *p.Spec.Thanos.Image != "" {
 			thanosImage = *p.Spec.Thanos.Image
 		}
+		thanosArgs := []string{
```

Comment:
Small nit: 

Why not just do:
```
                         Args: []string{	
				"sidecar",	
				fmt.Sprintf("--prometheus.url=http://%s:9090%s", c.LocalHost, path.Clean(webRoutePrefix)),	
				fmt.Sprintf("--tsdb.path=%s", storageDir),	
				fmt.Sprintf("--grpc-address=%s:10901", bindAddress),	
				fmt.Sprintf("--http-address=%s:10902",	bindAddress),
			},
```
and find proper `bindAddress`  beforehand? Might more consistent with what we had before and as a result bit more readable (:

Reply:
Thanks, good point, reads far better now!

In terms of `0.0.0.0` vs `127.0.0.1` I swayed towards the more limited change, and it also matched the existing behaviour in #1051 - as far as I'm aware there's no limit/issue caused by listening on `0.0.0.0` instead, and it does match the default Thanos behaviour. If we decide to go with that instead I'd like to rename the variable to be more accurate.

Reply:
Then let's go with `127.0.0.1`. We just won't be able to have any probes on those ports.
But there might be other ways to add them for those containers: https://github.com/coreos/prometheus-operator/pull/2763